src/fields/lib/abstract-field.js

```diff
@@ -301,20 +269,70 @@ module.exports = class AbstractField {
    * @returns {Promise.<boolean>}
    */
   canWriteOnUpdate(bundle) {
-    if (this._canWriteOnUpdate) {
-      return Promise.resolve(this._canWriteOnUpdate.call(null, bundle))
-        .then(result => {
-          if (!result) {
-            return Promise.reject(new Restypie.TemplateErrors.FieldNotUpdatable({
-              key: this.key
-            }));
-          }
-        });
+    if (this._canWriteOnUpdateField) {
+      return Promise.resolve(this._canWriteOnUpdateField.call(null, bundle));
     }
     return Promise.resolve(true);
   }
 
   /**
+   * Rejects the request by throwing if canRead returns false
+   *
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   * @private
+   */
+  _canRead(bundle) {
+    return Promise.resolve(this.canRead(bundle))
+      .then(result => {
+        if (!result) {
+          return Promise.reject(new Restypie.TemplateErrors.FieldNotReadable({
+            key: this.key
+          }));
+        }
+        return Promise.resolve(result);
+      });
+  }
+
+  /**
+   * Rejects the request by throwing if canWriteOnCreate returns false
+   *
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   * @private
+   */
+  _canWriteOnCreate(bundle) {
+    return Promise.resolve(this.canWriteOnCreate(bundle))
+      .then(result => {
+        if (!result) {
+          return Promise.reject(new Restypie.TemplateErrors.FieldNotWritable({
+            key: this.key
+          }));
+        }
+        return Promise.resolve(result);
+      });
+  }
+
+  /**
+   * Rejects the request by throwing if canWriteOnUpdate returns false
+   *
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   * @private
+   */
+  _canWriteOnUpdate(bundle) {
+    return Promise.resolve(this.canWriteOnUpdate(bundle))
```

Comment:
Why do you need to wrap this into `Promise.resolve` ? Doesn't `canWriteOnUpdate` return a Promise already ?